eng/pipelines/templates/jobs/ci.yml

```diff
@@ -229,11 +234,23 @@ jobs:
   - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
```

Comment:
We will want to duplicate this entire template if TestDependsOnDependency is not empty.

Reply:
I am currently passing different set of parameters based on `TestDependsOnDependency `. Why do we want duplicate the template instead?